library/src/main/scala/scala/scalajs/js/BigInt.scala

```diff
@@ -0,0 +1,188 @@
+package scala.scalajs.js
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSGlobal
+
+/**
+ * A built-in object that provides a way to represent whole numbers larger than
+ * 2 ^ 53 - 1, which is the largest number JavaScript can reliably represent
+ * with the Number primitive.
+ *
+ * BigInt can be used for arbitrarily large integers.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+ */
+@js.native
+@JSGlobal
+class BigInt private[this] () extends js.Object {
+
+  def +(other: BigInt): BigInt = js.native
+  def *(other: BigInt): BigInt = js.native
+  def -(other: BigInt): BigInt = js.native
+  def **(other: BigInt): BigInt = js.native
+  def %(other: BigInt): BigInt = js.native
+
+  def &(other: BigInt): BigInt = js.native
+  def |(other: BigInt): BigInt = js.native
+  def ^(other: BigInt): BigInt = js.native
+  def <<(other: BigInt): BigInt = js.native
+  def >>(other: BigInt): BigInt = js.native
+  def >>>(other: BigInt): BigInt = js.native
+  // no >>>
+
+  def unary_- : BigInt = js.native
+  def unary_~ : BigInt = js.native
+  // no unary_+
+
+  def ==(x: BigInt): Boolean = js.native
+  def ==(x: Short): Boolean = js.native
+  def ==(x: Char): Boolean = js.native
+  def ==(x: Int): Boolean = js.native
+  def ==(x: Long): Boolean = js.native
+  def ==(x: Float): Boolean = js.native
+  def ==(x: Double): Boolean = js.native
+
+  def !=(x: BigInt): Boolean = js.native
+  def !=(x: Byte): Boolean = js.native
+  def !=(x: Short): Boolean = js.native
+  def !=(x: Char): Boolean = js.native
+  def !=(x: Int): Boolean = js.native
+  def !=(x: Long): Boolean = js.native
+  def !=(x: Float): Boolean = js.native
+  def !=(x: Double): Boolean = js.native
+
+  def <(x: BigInt): Boolean = js.native
+  def <(x: Byte): Boolean = js.native
+  def <(x: Short): Boolean = js.native
+  def <(x: Char): Boolean = js.native
```

Comment:
Comparisons with `Char`s and `Long`s will not work. They should not be declared.

In general, I don't think we should expose the comparisons with other types than `BigInt` and `Double`. And even `Double` is border-line, because we're not exposing the converse comparisons where `Double` is on the left and `BigInt` is on the right. I believe it's better to leave them out.

Reply:
I agree to expose comparison only for `BigInt` and `Double`.
But I do not understand why `Double op BigInt` should not be exposed.

Reply:
Well we could expose `Double op BigInt`, yes, but currently your PR doesn't do that. It only exposes `BigInt op Double`. I'm against exposing one direction but not the other. It should be "commutative".